### PR TITLE
New add/edit supply dialog

### PR DESCRIPTION
**Pull request by QuicheLorraine3600**
This PR removes the SupplyPopup and replaces it by EditSupplyDialog. This new Dialog is meant to be used both in the EventCreationScreen and the SuppliesScreen (future PR).

Here you can have a look of this new dialog:
image

This dialog uses the CustomDialog class, which is just a Dialog with a title, a description, a content and some buttons.

### CODE REVIEW

File: app/src/main/java/com/monkeyteam/chimpagne/ui/event/details/supplies/EditSupplyDialog.kt
Comment (Kuriv2001) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/event/details/supplies/EditSupplyDialog.kt:1]: Missed value: KeyboardActions(onDone = { /* Handle Done action */}),?

**Review by Kuriv2001 — COMMENTED** [review]
Well done!

The new screen looks better and works well.

I ran into some problem when putting three or more digits for the quantity of the supplies, then the number is not correctly displayed. For example when i put 100 in the quantity field i see this:

Screenshot 2024-05-13 at 13 50 06
But then when i click okay i get the 100:
Screenshot 2024-05-13 at 13 50 12

The error handling looks very well done and is foolproof.

**Comment by QuicheLorraine3600** [comment]
> Well done!
> 
> The new screen looks better and works well.
> 
> I ran into some problem when putting three or more digits for the quantity of the supplies, then the number is not correctly displayed. For example when i put 100 in the quantity field i see this:
> Screenshot 2024-05-13 at 13 50 06
> 
> But then when i click okay i get the 100: Screenshot 2024-05-13 at 13 50 12
> 
> The error handling looks very well done and is foolproof.

Typical case of "it's working on my device" and example of non responsiveness :sob:. I will fix it

File: app/src/main/java/com/monkeyteam/chimpagne/ui/event/details/supplies/EditSupplyDialog.kt
Comment (QuicheLorraine3600) [reply app/src/main/java/com/monkeyteam/chimpagne/ui/event/details/supplies/EditSupplyDialog.kt:1]: This code is from @Athanaze, I think I should remove it

**Review by Kuriv2001 — APPROVED** [review]
Well done! Looks all good now

**Pull request merged**